packages/itwin/grouping-mapping-widget/src/components/Groups/GroupsVisualization.tsx

```diff
@@ -105,24 +105,44 @@ export const GroupsVisualization = ({
   const groupQueriesProgressCount = useMemo(() => groupQueries.filter((query) => query.isFetched).length, [groupQueries]);
   const isResolvingGroupQueries = useMemo(() => groupQueries.some((query) => query.isFetching), [groupQueries]);
 
-  const hiliteIds = useMemo(
-    () =>
-      isGroupsQueriesReady
-        ? groupQueries.map((query) => ({
-          groupId: query.data!.group.id,
-          elementIds: query.data!.result.ids,
-        }))
-        : [],
-    [groupQueries, isGroupsQueriesReady]
-  );
+  const hiliteIds = useMemo(() => {
+    if (!isGroupsQueriesReady || !groups) return [];
+
+    const processedGroupIds = new Map<string, string[]>();
+
+    return groupQueries.flatMap((query) => {
+      // Find all groups that match the current query and haven't been processed yet
```

Comment:
What does `processed` mean here?

Reply:
I wrote a comment above. I guess associated would be a better term. Processed means associated with hiliteid results. This whole logic exists to avoid issues when their are groups with the same query.